src/patchelf.cc

```diff
@@ -787,10 +801,51 @@ void ElfFile<ElfFileParamNames>::writeReplacedSections(Elf_Off & curOff,
             }
         }
 
-        curOff += roundUp(i->second.size(), sectionAlignment);
+        curOff += roundUp(i->second.data.size(), sectionAlignment);
     }
 
-    replacedSections.clear();
+    /* Expand the LOAD segment as neccessary */
```

Comment:
This is the extra mandatory fix -- in fact, I think that's a regression from v0.15, originally posted here:

https://github.com/NixOS/patchelf/pull/264

Without this fix, `libnode.so` is broken to the point where it crashes inside the linker :smile: